tests/components/tessie/test_config_flow.py

```diff
@@ -137,3 +140,131 @@ async def test_form_network_issue(hass: HomeAssistant) -> None:
             TEST_CONFIG,
         )
     assert result3["type"] == FlowResultType.CREATE_ENTRY
+
+
+async def test_reauth(hass: HomeAssistant) -> None:
+    """Test reauth flow."""
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_CONFIG,
+        unique_id="abc",
+    )
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "unique_id": mock_entry.unique_id,
+            "entry_id": mock_entry.entry_id,
+        },
+        data=TEST_CONFIG,
+    )
+
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "reauth_confirm"
+    assert not result["errors"]
+
+    with patch(
+        "homeassistant.components.tessie.config_flow.get_state_of_all_vehicles",
+        return_value=TEST_STATE_OF_ALL_VEHICLES,
+    ) as mock_get_state_of_all_vehicles, patch(
+        "homeassistant.components.tessie.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            TEST_CONFIG,
+        )
+        await hass.async_block_till_done()
+        assert len(mock_setup_entry.mock_calls) == 1
+        assert len(mock_get_state_of_all_vehicles.mock_calls) == 1
+
+    assert result2["type"] == FlowResultType.ABORT
+    assert result2["reason"] == "reauth_successful"
+    assert mock_entry.data == TEST_CONFIG
+
+
+async def test_reauth_error_auth(hass: HomeAssistant) -> None:
```

Comment:
Optional suggestion: You could parameterize these tests with the side effect and expected error to combine the last 3 tests, since i think they are equivalent otherwise.

Reply:
Took your advice and have parameterize the last 3 tests.